doc/sphinx/practical-tools/coq-commands.rst

```diff
@@ -226,6 +233,89 @@ and ``coqtop``, unless stated otherwise:
+At this point, one is ready to work interactively on the file ``foo.v``, even
+though it was never needed to compile the proofs involved in the ``bar*.v``
+files.
+
+Assume a set of files ``f1.v ... fn.v`` with linear dependencies. The command
+``make vos`` enables compiling the statements (i.e. excluding the proofs) in all
+the files. Next, ``make -j vok`` enables compiling all the proofs in parallel.
+Thus, calling ``make -j vok`` directly enables taking advantage of a maximal
+amount of parallelism during the compilation of the set of files.
+
+Note that this comes at the cost of parsing and typechecking all definitions
+twice, once for the ``.vos`` file and once for the ``.vok`` file. However, if
+files contain nontrivial proofs, or if the files have many linear chains of
+dependencies, or if one has many cores available, compilation should be faster
+overall.
+
+**Need for ``Proof using``**
+
+When a theorem is part of a section, typechecking the statement of this theorem
+might be insufficient for deducing the type of this statement as of at the end
+of the section. Indeed, the proof of the theorem could make use of section
+variables or section hypotheses that are not mentioned in the statement of the
+theorem.
+
+For this reason, proofs inside section should begin with :cmd:`Proof using`
+instead of ``Proof``, where after the ``using`` one should provide the list of
+the names of the section variables that are required for the proof but are not
+involved in the typechecking of the statement. Note that it is safe to write
+``Proof using.`` instead of ``Proof.`` also for proofs that are not within a
+section.
+
+.. warn:: You should use the “Proof using [...].” syntax instead of “Proof.” to
+enable skipping this proof which is located inside a section. Give as argument
+to “Proof using” the list of section variables that are not needed to typecheck
+the statement but that are required by the proof.
+
+     If |Coq| is invoked using the ``-vos`` option, whenever it finds the
+     command ``Proof.`` inside a section, it will compile the proof, that is,
+     refuse to skip it, and it will raise a warning. To disable the warning, one
+     may pass the flag ``-w -proof-without-using-in-section``.
+
+**Interaction with standard compilation**
+
+When compiling a file ``foo.v`` using |coqc| in the standard way (i.e., without
+``-vos`` nor ``-vok``), an empty file ``foo.vos`` is created in addition to the
+regular output file ``foo.vo``. If |coqc| is subsequently invoked on some other
+file ``bar.v`` using option ``-vos`` or ``-vok``, and that ``bar.v`` requires
+``foo.v``, if |Coq| finds an empty file ``foo.vos``, tyen it will load
+``foo.vo`` instead of ``foo.vos``.
+
+The purpose of this feature is to allow users to benefit from the ``-vos``
+option even if they depend on libraries that were compiled in the traditional
+manner (i.e., never compiled using the ``-vos`` option).
```

Comment:
What about being able to read the `.vo` files even if the `.vos` file is not present? Or is that something that `make` wouldn't be able to support?

Reply:
It might seem like a good idea at first, but I think it is going to raise a lot of confusion in case the user somehow screws up his Makefile and his lacking dependencies on .vos files. Then, he will have no clue what is going on when he gets errors of the form "interface foo is more recent than interface bla" in the compilation of a subsequent file. Having an explicit empty ".vos" file has some cost (it pollutes your "ls"), but it enables keeping the semantics simple.

Reply:
Indeed, I can see the point.